app/src/main/java/com/example/sarahtang/mdbsocials/DetailsActivity.java

```diff
@@ -0,0 +1,220 @@
+                Log.d("GotImageFromStorage", uri.toString());
+            }
+        }).addOnFailureListener(new OnFailureListener() {
+            @Override
+            public void onFailure(@NonNull Exception exception) {
+                Log.d("NoImageFromStorage", exception.toString());
+            }
+        });
+
+        //Setting text for each Text view in layout
+        nameDetail.setText(name);
+        hostDetail.setText(getString(R.string.hostedBy) + host);
+        descriptionDetail.setText(getString(R.string.descriptionIntro) + description);
+
+        switch(numberInterested) {
+            case 1:
+                interestedDetail.setText(numberInterested + getString(R.string.personInterested));
+                break;
+            default:
+                interestedDetail.setText(numberInterested + getString(R.string.peopleInterested));
+        }
+    }
+
+    /**
+     * Transaction to increase number of people interested by one and to add person to
+     * interested list (if not already added);
+     * Modularized ish
+     */
+    public void addInterested() {
+        Log.d("TRANSACTION", "Success");
+
+        final DatabaseReference ref = FirebaseDatabase.getInstance().getReference("/socials");
+        ref.child(firebaseKey).runTransaction(new Transaction.Handler() {
+            @Override
+            public Transaction.Result doTransaction(MutableData mutableData) {
+                //String currEmail = FirebaseAuth.getInstance().getCurrentUser().getEmail();
+
+                if (!buttonPressed) {
+                    buttonPressed = true;
+                    int numInterested = mutableData.child("numberInterested").getValue(Integer.class);
+                    numInterested++;
+                    Log.d("Details", "transaction");
+                    DetailsActivity.addInterestedToDatabase(numInterested,firebaseKey);
+
+
+
+                    ArrayList<String> a = (ArrayList) mutableData.child("peopleInterested").getValue();
+                    if (a == null) {
+                        a = new ArrayList<String>();
+                    }
+
+                    if (!a.contains(currEmail)) {
+                        a.add(currEmail);
+                        ref.child(firebaseKey).child("peopleInterested").setValue(a);
+                    }
+
+                }
+                else {
+                    Toast.makeText(getApplicationContext(), R.string.toastAlreadyInterested, Toast.LENGTH_SHORT).show();
+                }
+
+                return Transaction.success(mutableData);
+            }
+            @Override
+            public void onComplete(DatabaseError databaseError, boolean b, DataSnapshot dataSnapshot) {
+
+            }
+        });
+    }
+
+    public static void addInterestedToDatabase(int interested, String firekey) {
+        DatabaseReference ref = FirebaseDatabase.getInstance().getReference("/socials");
+        ref.child(firekey).child("numberInterested").setValue(interested);
```

Comment:
These types of strings are important to keep as final variables to ensure consistency. You can get a String wrong in another class, but you can't get a variable name wrong and still compile